de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/AggregationEngine.java

```diff
@@ -540,12 +541,15 @@ private int getBill_User_ID(@NonNull final I_C_Invoice_Candidate ic, final boole
 
 		final RetrieveContactRequest request = RetrieveContactRequest
 				.builder()
+				.forceActive(true)
+				.contactType(ContactType.BILL_TO_DEFAULT)
+				.forceType(true)
```

Comment:
btw, `isUpdateLocationAndContactForInvoice` is not the right name....

Reply:
also some unit tests would be very nice

Reply:
Used useDefaultBillLocationAndContactIfNotOverride.

There already were some unit tests on this functionality but they did not cover the cases in this issue. Added them: de.metas.invoicecandidate.api.impl.aggregationEngine.TestUpdatedLocationAndUser